tests/kafkatest/tests/core/snapshot_test.py

```diff
@@ -0,0 +1,224 @@
+        # Wait until the first log segment in metadata log is marked for deletion
+        wait_until(lambda: not self.file_exists(node, file_path),
+                   timeout_sec=100,
+                   backoff_sec=1, err_msg="Not able to verify cleanup of log file %s in a reasonable amount of time" % file_path)
+
+    def wait_for_snapshot(self, node):
+        # Wait for a snapshot file to show up
+        file_path = self.kafka.METADATA_SNAPSHOT_SEARCH_STR
+        wait_until(lambda: self.file_exists(node, file_path),
+                   timeout_sec=100,
+                   backoff_sec=1, err_msg="Not able to verify snapshot existence in a reasonable amount of time")
+
+    def file_exists(self, node, file_path):
+        # Check if the first log segment is cleaned up
+        self.logger.debug("Checking if file %s exists" % file_path)
+        cmd = "ls %s" % file_path
+        files = node.account.ssh_output(cmd, allow_fail=True, combine_stderr=False)
+
+        if len(files) is 0:
+            self.logger.debug("File %s does not exist" % file_path)
+            return False
+        else:
+            self.logger.debug("File %s was found" % file_path)
+            return True
+
+    def validate_success(self, topic = None):
+        if topic is None:
+            # Create a new topic
+            topic = "%s%d" % (TestSnapshots.TOPIC_NAME_PREFIX, self.topics_created)
+            self.topics_created += self.create_n_topics(topic_count=1)
+
+        # Produce to the newly created topic to ensure broker has caught up
+        self.producer = VerifiableProducer(self.test_context, self.num_producers, self.kafka,
+                                           topic, throughput=self.producer_throughput,
+                                           message_validator=is_int)
+
+        self.consumer = ConsoleConsumer(self.test_context, self.num_consumers, self.kafka,
+                                        topic, consumer_timeout_ms=30000,
+                                        message_validator=is_int)
+        self.start_producer_and_consumer()
+        self.stop_producer_and_consumer()
+        self.validate()
+
+    @cluster(num_nodes=9)
+    @matrix(metadata_quorum=quorum.all_kraft)
+    def test_broker(self, metadata_quorum=quorum.colocated_kraft):
+        # The following set of scenarios are meant to test the ability
+        # of a broker to consume metadata snapshots and to recover using them
+        #
+        # During setup phase, the test ensures that there is atleast
+        # one snapshot created on the controllers and that at least the first
+        # log segment has been truncated (requiring an observer to always load 
+        # a snapshot to catch up to the metadata state)
+        #
+        # Each scenario is a progression over the previous one.
+        # Redundant scenarios are left in to make debugging a failure of the test easier
+
+        # Scenario -- Re-init broker after cleaning up all persistent state
```

Comment:
Should this be a separate test function? Putting it here seems like it might muddy the waters, especially if the nodes you hit are combined controller/brokers.